notifications-sw.js

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * UsersManager screenshot tests.
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+var token;
+
+self.addEventListener('install', async function() {
+    self.token = new URL(location).searchParams.get('token');
+});
+
+self.addEventListener('activate', async function() {
+    // Fetch any notifications that have already been generated
+    fetchNotifications();
+
+    // Check periodically for new notifications
+    this.timerId = setInterval(function() {
+        fetchNotifications();
+    }, 30 * 1000);  // Timeout is in ms
+});
+
+var fetchNotifications = function() {
+    fetch('index.php?module=MobileMessaging&action=getBrowserNotifications&token=' + self.token, 
+        {credentials: "omit"}
+    ).then(function(response) {
+        if (response.status === 200) {
+            displayNotifications(response);
+        }
+    });
+};
+
+var displayNotifications = function(response) {
+    response.json().then(function(notifications) {
+        notifications.forEach(function(notification) {
```

Comment:
It may be good to check that notifications is an array. We've had random problems there sometimes in the past